lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -2750,16 +2749,76 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
               binder.op, "unimplemented: cubic coeff must be -0.75");
         }
 
-        unsigned rank = dyn_cast<Torch::ValueTensorType>(operands[0].getType())
-                            .getSizes()
-                            .size();
+        Value inputTensor = operands[0];
+        Torch::ValueTensorType inputTensor_type =
+            cast<Torch::ValueTensorType>(inputTensor.getType());
+        ArrayRef<int64_t> inputTensor_sizes = inputTensor_type.getSizes();
+        ArrayRef<int64_t> outputTensor_sizes = outputTensor_type.getSizes();
+
+        int64_t const batchDimension = 0;
+        int64_t const channelDimension = 1;
+        int64_t nonScalableDimensions[] = {
+            batchDimension,
+            channelDimension,
+        };
+
+        auto errorMessageForScaling = [](int64_t givenDimension) {
+          switch (givenDimension) {
+          case batchDimension:
+            return "Unexpected intent to scale the batch dimension";
+          case channelDimension:
+            return "Unexpected intent to scale the channel dimension";
+          default:
+            return "Scalable dimension treated as non-scalable";
+          }
+        };
+
+        auto unknownSize = Torch::kUnknownSize;
+
+        // Compile-time check for dimensions of static size
+        for (auto eachDimension : nonScalableDimensions) {
```

Comment:
This is a personal preference, but I would certainly prefer:
```suggestion
        // Compile-time check for dimensions of static size
        for (int64_t i = 0; i<2; i++) {
```

This is for two reasons: 

1. Although the torch op might have these specific dimension specifications, the ONNX op does not. We should say something a bit more generic like "unsupported: non-trivial scaling in the first two dimensions.". 
2. We might want to implement support for this path instead of writing a match failure in the future. With all of the variables and structures introduced here, I think this would cause more work for the future developer.

Reply:
And personal preferences are worth sharing! They're heuristics for our engineering experiences, and they're worth putting out there because:
- they could help someone solve a problem that they've encountered in their own experience
- it's an opportunity to evaluate a preference, put them to words, recalibrate them, or maybe even drop them altogether when presented with new information.

1. Gotcha, so we want the error message to not reveal the rational. In that case, I'll have the message report the dimension index instead. Also gives us a chance to differentiate the messages between compile-time and run-time check.
2. I agree that it's work! In the case of adding support, work is unavoidable:
   - the work of deleting 3 variable declarations that contain cognitive load
     OR
   - the work of gathering the required cognitive load by
     - talking to other engineers
     - researching within the codebase
     - reading external docs
     - etc.
    OR
   - some other unforeseen work!

Predicted questions by future devs (especially if they're also cross-discipline):
- "Wait, why 2 loops? What's significant about the first two dimensions?"
  - *pings engineer*
- "Okay, so they shouldn't be scaled, but why? What do they represent?"
  - *pings engineer again*
- "Ah, gotcha, they're the batch and channels dims, the elements in which have nothing to do spatial dimensions"

With the dims explicitly declared and the implementation based on them, another engineer might be able to avoid having to recollect the CL we had already collected once before.